BEP18.md

```diff
@@ -0,0 +1,123 @@
+
+This BEP introduces the following details:
+
+- What's the procedure to take a snapshot
+- What's the procedure to sync snapshot from other peers
+- Snapshot (manifest, snapshot chunks) format
+
+## 3.  Status
+
+This BEP is still working in progress.
+
+## 4.  Motivation
+
+We propose this BEP to enhance full node user experience (and ease their pain) on using state sync because of following implementation limitations.
+
+1. Users complain most about state syncing testnet is very slow and usually stuck on some requests. <br> <br> In this enhancement, we want data to respond more evenly across peers so that syncing can continuously make progress and the overall syncing time can reduce from 30 - 45 min to around 5 min.
+
+2. Interruption during state sync (node process get killed because of reboot computer or user impatience) would make already synced data in vain (because the current full node doesn't persist synced part on disk). Worsely it mistakenly writes a lock file prevents user state sync again. <br> <br> In this enhancement, we want support break-resume downloading and keep the consistent status for arbitrarily restart.
+
+## 5.  Specification
+
+State sync will download **manifest** and **snapshot chunks** from other peers.
+
+### 5.1 Take snapshot
+
+There are two ways to take snapshots from a fullnode: automatically or manually. Snapshots will be put under `$HOME/data/snapshot/<height>`. All types involved in the snapshot are encoded by go-amino and compressed by snappy. More details will be explained later.
+
+1. To make fullnode automatically take snapshots, just make sure `state_sync_reactor` in `$HOME/config.toml` is set to true.
+2. To manually take snapshots, stop the node if it is running, then run `./bnbchaind snapshot --home <home> --height <height>`.
+
+If the snapshot taking procedure is interrupted, the node will be still in good status, but it cannot provide the interrupted height for other peers.
+
+Note: automatically take snapshot would keep occupying disk space. Fullnode would not delete them automatically, so the user should periodically delete unneeded snapshots manually if they want to save disk space.
+
+### 5.2 Sync snapshot
+
+Syncing snapshot is designed to be only run once during full node first start. To enable state sync from others, `state_sync_reactor` should be true and `state_sync_height` should be set to non-negative (default `-1` means disable syncing from others).
+
+If a user wants to sync from (majority) peers' latest sync-able height, they should set `state_sync_height` to 0.
+
+Stop and restart fullnode during state sync is allowed. The next time full node is started, it will resume by loading Manifest and downloaded chunks then download missing chunks.
+
+Once state sync is successful, a `STATESYNC.LOCK` file will be created under `$HOME/data` to prevent state sync next time.
```

Comment:
Does this mean node can only sync once?

Reply:
> Does this mean node can only sync once?

nodes are supposed to only sync once. But as I wrote in this BEP, they can delete this file manually and trigger state sync again if they are too many days lagged. (but their `blockchain.db` is not continuous in this case)